src/bp/core/routers/bots/index.ts

```diff
@@ -281,15 +282,51 @@ export class BotsRouter extends CustomRouter {
     )
 
     this.router.post(
-      '/flows',
+      '/flow',
       this.checkTokenHeader,
       this.needPermissions('write', 'bot.flows'),
       this.asyncMiddleware(async (req, res) => {
-        const botId = req.params.botId
-        const flowViews = <FlowView[]>req.body.dirtyFlows
+        const { botId } = req.params
+        const flow = <FlowView>req.body.flow
 
-        await this.flowService.saveAll(botId, flowViews, req.body.cleanFlows)
-        res.sendStatus(201)
+        await this.flowService.insertFlow(botId, flow)
+
+        res.sendStatus(200)
+      })
+    )
+
+    this.router.put(
+      '/flow/:flowName',
+      this.checkTokenHeader,
+      this.needPermissions('write', 'bot.flows'),
+      this.asyncMiddleware(async (req, res) => {
+        const { botId, flowName } = req.params
+        const flow = <FlowView>req.body.flow
+
+        if (_.has(flow, 'name') && flowName !== flow.name) {
+          await this.flowService.renameFlow(botId, flowName, flow.name)
+        } else {
+          await this.flowService.updateFlow(botId, flow)
+        }
+        res.sendStatus(200)
+      })
+    )
+
+    this.router.delete(
+      '/flow/:name',
+      this.checkTokenHeader,
+      this.needPermissions('write', 'bot.flows'),
+      this.asyncMiddleware(async (req, res) => {
+        const { botId, flowName } = req.params
+
+        if (!_.isString(flowName) && flowName.length) {
```

Comment:
why is `flowName.length` needed ?

Reply:
You're right there's no need for it. In fact, there's no need for !_.isString(flowName) as a path param is necessarily a string.